### PR TITLE
Added PaymentIntents API

### PR DESCRIPTION
**Pull request by stevehayles**
Expose additional calls to allow using the library with the new PaymentIntents api

Small refactor to clarify the calls into the underlying Stripe library

### CODE REVIEW

**Comment by paurosello** [comment]
Do you think we can get this merged any time soon? :)

**Comment by leecrosdale** [comment]
A bit concerned that the last code change was 10 months ago.  Would happily take over the repo if the dev doesn't have time to run it anymore. Would rather get code like this in sooner rather than later.

**Comment by paurosello** [comment]
I had problems with latest version of webpack while compiling this branch so more work will have to be done I think, my solution is bad as I used version 2.x in order to make it build

**Comment by mccare** [comment]
I've created a pull request (https://github.com/fromAtoB/vue-stripe-elements/pull/75) and adapted webpack.config.js so it will build with webpack 4.

**Comment by mydnic** [comment]
@stevehayles isn't the `createPaymentMethod `method required as well ? https://stripe.com/docs/stripe-js/reference#stripe-create-payment-method

In their migration guide they suggest to switch from `createToken `to `createPaymentMethod`
(https://stripe.com/docs/payments/payment-intents/migration#web)

File: src/stripeElements.js [hunk 1]
```diff
@@ -47,9 +52,14 @@ export function create(elementType, key_or_stripe, options = {}) {
...
-  Stripe.createSource = (options) => Stripe.instance.createSource(element, options)
-  Stripe.retrieveSource = (options) => Stripe.instance.retrieveSource(options)
+  Stripe.createToken = (tokenData) => Stripe.instance.createToken(element, tokenData)
+  Stripe.createSource = (sourceData) => Stripe.instance.createSource(element, sourceData)
+  Stripe.retrieveSource = (source) => Stripe.instance.retrieveSource(source)
+  Stripe.paymentRequest = (options) => Stripe.instance.paymentRequest(options)
+  Stripe.redirectToCheckout = (options) => Stripe.instance.redirectToCheckout(options)
+  Stripe.retrievePaymentIntent = (clientSecret) => Stripe.instance.retrievePaymentIntent(clientSecret)
+  Stripe.handleCardPayment = (clientSecret, data) => Stripe.instance.handleCardPayment(clientSecret, element, data)
+  Stripe.confirmPaymentIntent = (clientSecret, data) => Stripe.instance.confirmPaymentIntent(clientSecret, element, data)
```
[the remark this answers is not in the document]
Comment (mydnic) [reply src/stripeElements.js]: And https://stripe.com/docs/stripe-js/reference#stripe-handle-card-action ?

[the remark this answers is not in the document]
Comment (stevehayles) [reply src/stripeElements.js]: The 'createPaymentMethod' call is not required for the automatic confirmation flow which is the cleanest way to use the Stripe API.  I agree though that it's still a useful call and should be added. 

I elected not to use this library in the end and haven't updated it but I will look at adding the extras calls.

[the remark this answers is not in the document]
Comment (richie3366) [reply src/stripeElements.js]: Hello @stevehayles, I was going to use `createPaymentMethod`, but I'm concerned about doing it right. Could you please tell us which method to call? (precision: I'm in a SetupIntent scenario)
Thanks

[the remark this answers is not in the document]
Comment (nickdnk) [reply src/stripeElements.js]: > The 'createPaymentMethod' call is not required for the automatic confirmation flow which is the cleanest way to use the Stripe API. I agree though that it's still a useful call and should be added.
> 
> I elected not to use this library in the end and haven't updated it but I will look at adding the extras calls.

createPaymentMethod is definitely necessary for the automatic flow. You don't need to decide on which flow you want to use until you create a PaymentIntent.

[the remark this answers is not in the document]
Comment (stevehayles) [reply src/stripeElements.js]: I'm processing payments just fine without it and the docs are clear you don't need it for automatic flow as are the docs for other libraries like 'react-stripe-elements'.  

I am no longer using this library and can't test any further changes easily but I have just committed some changes that will add the calls (https://github.com/stevehayles/vue-stripe-elements/commit/07cb1c22d98cb250fec64833cec80425a2ae0ecd)

**Comment by mydnic** [comment]
Is anyone from fromAtoB available to merge this ?
Poking @softbeehive

File: src/stripeElements.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickdnk) [reply src/stripeElements.js]: > I'm processing payments just fine without it and the docs are clear you don't need it for automatic flow as are the docs for other libraries like 'react-stripe-elements'.
> 
> I am no longer using this library and can't test any further changes easily but I have just committed some changes that will add the calls ([stevehayles@07cb1c2](https://github.com/stevehayles/vue-stripe-elements/commit/07cb1c22d98cb250fec64833cec80425a2ae0ecd))

It depends on what you're doing. If it's something complex it quickly becomes useful, automatic or not. So no, perhaps necessary was the wrong word, but it certainly is not something that should be left out of a library like this.

**Comment by richie3366** [comment]
Would be lovely if you could also add the `handleCardSetup` method, needed in the recent SetupIntents API (off-session payments).

https://stripe.com/docs/stripe-js/reference#stripe-handle-card-setup

**Comment by mydnic** [comment]
@stevehayles you mentioned you were using another solution, may I ask which one ? Thanks!

**Comment by damarinov1** [comment]
@softbeehive Do you think we can get this merged any time soon? :)

**Comment by VladZen** [comment]
God bless you @stevehayles! It's very smart to overload Stripe Instance's methods, but until this will be merged I see no other way besides using exported `instance` and handle new methods manually.

**Pull request merged**